plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.block_theme.spec.ts

```diff
@@ -966,6 +966,63 @@ test.describe( 'Product Collection', () => {
 				.poll( async () => await page.evaluate( 'window.eventFired' ) )
 				.toBe( 3 );
 		} );
+
+		test.describe( 'wc-blocks_viewed_product is emitted', () => {
+			let promise: Promise< { productId?: number; collection?: string } >;
+
+			test.beforeEach( async ( { page, pageObject } ) => {
+				await pageObject.createNewPostAndInsertBlock( 'featured' );
+
+				promise = new Promise( ( resolve ) => {
+					void page.exposeFunction( 'resolvePayload', resolve );
+					void page.addInitScript( () => {
+						window.document.addEventListener(
+							'wc-blocks_viewed_product',
+							( e ) => {
+								window.resolvePayload( e.detail );
+							}
+						);
+					} );
+				} );
+
+				await pageObject.publishAndGoToFrontend();
+			} );
+
+			test( 'when Product Image is clicked', async ( { page } ) => {
+				await page
+					.locator( '[data-block-name="woocommerce/product-image"]' )
+					.nth( 0 )
+					.click();
+
+				const { collection, productId } = await promise;
+				expect( collection ).toEqual(
+					'woocommerce/product-collection/featured'
+				);
+				expect( productId ).toEqual( expect.any( Number ) );
+			} );
+
+			test( 'when Product Title is clicked', async ( { page } ) => {
+				await page.locator( '.wp-block-post-title' ).nth( 0 ).click();
+
+				const { collection, productId } = await promise;
+				expect( collection ).toEqual(
+					'woocommerce/product-collection/featured'
+				);
+				expect( productId ).toEqual( expect.any( Number ) );
+			} );
+
+			test( 'when Add to Cart Anchor is clicked', async ( { page } ) => {
+				await page
+					.getByLabel( 'Select options for “V-Neck T-' )
+					.click();
+
+				const { collection, productId } = await promise;
+				expect( collection ).toEqual(
+					'woocommerce/product-collection/featured'
+				);
+				expect( productId ).toEqual( expect.any( Number ) );
+			} );
+		} );
```

Comment:
What do you think if we move these tests in a dedicated file like `product-collection-tracking.block_theme.spec.ts`?

Reply:
Good point! Done in https://github.com/woocommerce/woocommerce/pull/51156/commits/41c645cebe06a1d97018148e7a1c1b080fdc8d7f